frontend/packages/noobaa-storage-plugin/src/components/buckets-card/buckets-card-item.tsx

```diff
@@ -1,75 +1,114 @@
+      </Link>
+    </div>
+  );
+});
+
+// Displays count of erroneous buckets
+const BucketFailureStatus: React.FC<{ count: number[]; link: string[] }> = React.memo(
+  ({ count, link }) => (
+    <div className="nb-buckets-card__buckets-failure-status">
+      {count[0] > 0 && <OtherFailure link={link[0]} status={count[0]} />}
+      {count[1] > 0 && <ProvisioningFailure link={link[1]} status={count[1]} />}
+    </div>
+  ),
+);
+
+// Displays count of buckets and objects present in buckets
+const BucketsStatus: React.FC<BucketsStatusProps> = React.memo(
+  ({ isLoading, hasLoadError, title, bucketsCount, objectsCount }) => {
+    let body: JSX.Element;
+    if (isLoading && !hasLoadError) {
+      body = (
+        <>
+          <div className="co-inventory-card__item-title">
+            <div className="skeleton-inventory" />
+            <div>{title}</div>
+          </div>
+          <div className="skeleton-text" />
+        </>
+      );
+    } else {
+      body = (
+        <>
+          <div>{hasLoadError ? title : pluralize(bucketsCount, title)}</div>
+          <div className="co-dashboard-text--small text-secondary">
+            {hasLoadError || !objectsCount ? 'Not available' : formatCount(Number(objectsCount))}
+          </div>
+        </>
+      );
+    }
+    return <div className="nb-buckets-card__buckets-status-title">{body}</div>;
   },
 );
 
+// Displays Buckets and Bucket Claims information as rows
 export const BucketsItem: React.FC<BucketsItemProps> = React.memo(
-  ({ title, bucketsCount, objectsCount, unhealthyCount, isLoading, link, error }) =>
-    isLoading ? (
-      <LoadingInline />
-    ) : (
-      <div className="co-inventory-card__item">
-        <BucketsRow title={title} bucketsCount={bucketsCount} objectsCount={objectsCount} />
-        <BucketsRowStatus status={unhealthyCount} link={link} error={error} />
-      </div>
-    ),
+  ({ isLoading, hasLoadError, title, bucketsCount, objectsCount, unhealthyCount, link }) => (
+    <div className="co-inventory-card__item">
+      <BucketsStatus
+        title={title}
+        bucketsCount={bucketsCount}
+        objectsCount={objectsCount}
+        hasLoadError={hasLoadError}
+        isLoading={isLoading}
+      />
+      {!(isLoading || hasLoadError) && <BucketFailureStatus count={unhealthyCount} link={link} />}
+    </div>
+  ),
 );
 
-export type BucketsType = {
-  bucketsCount: string;
+export type Buckets = {
+  bucketsCount: number;
+  hasLoadError: boolean;
   isLoading: boolean;
   objectsCount: string;
-  unhealthyCount: string | number;
-  error: boolean;
+  unhealthyCount: number[];
 };
 
-type BucketsItemProps = BucketsType & {
-  link: string;
+type BucketsItemProps = Buckets & {
   title: string;
+  link: string[];
 };
 
-type BucketsRowProps = {
-  bucketsCount: string;
-  title: string;
+type BucketsStatusProps = {
+  bucketsCount: number;
+  isLoading: boolean;
+  hasLoadError: boolean;
   objectsCount: string;
```

Comment:
why `bucketsCount` is typed as a number but not `objectsCount`. It would be nice to have coherency between the types of the props.

Reply:
`objectsCount` needs raw guage value to know whether they have some value or not. If not, then show "Not available".
This is not required in case of buckets count.